src/main/java/org/reaktivity/nukleus/kafka/internal/memory/MemoryLayout.java

```diff
@@ -128,67 +125,121 @@ public Builder minimumBlockSize(
         }
 
         public Builder maximumBlockSize(
-            int maximumBlockSize)
+            long maximumBlockSize)
         {
             if (!isPowerOfTwo(maximumBlockSize))
             {
                 throw new IllegalArgumentException("maximum block size MUST be a power of 2");
             }
 
+            if (maximumBlockSize >> Integer.numberOfTrailingZeros(ONE_GB) > Integer.MAX_VALUE)
+            {
+                throw new IllegalStateException("capacity too large, number of 1GB buffers would exceed Integer.MAX_VALUE");
+            }
+
```

Comment:
`maximumBlockSize` also cannot be greater than `1GB` due to implementation constraints, right?

Reply:
The maximum capacity that can be acquired is 1GB, because of the current implementation constraint.  As explained above I decided to preserve the " total memory" (or  overall capacity) meaning of maximumBlockSize.  Basically in the current code it has a double function, so I had to pick one.